src/configuration.ts

```diff
@@ -124,8 +131,28 @@ export function loadConfigurationFromPath(configFilePath: string) {
         let fileData = fs.readFileSync(configFilePath, "utf8");
         // remove BOM if present
         fileData = fileData.replace(/^\uFEFF/, "");
-        return JSON.parse(fileData);
+        const configFile: IConfigFile = JSON.parse(fileData);
+        configFile.rulesDirectory = getRulesDirectories(configFile.rulesDirectory, path.dirname(configFilePath));
+        return configFile;
+    }
+}
+
+export function extendConfigFile(config: IConfigFile, baseConfig: IConfigFile): IConfigFile {
+    let combinedConfig: IConfigFile = {};
+
+    const baseRulesDirectory = arrayify(baseConfig.rulesDirectory);
+    const configRulesDirectory = arrayify(config.rulesDirectory);
+    combinedConfig.rulesDirectory = configRulesDirectory.concat(baseRulesDirectory);
+
+    combinedConfig.rules = {};
+    for (const name of Object.keys(objectify(baseConfig.rules))) {
+        combinedConfig.rules[name] = baseConfig.rules[name];
+    }
+    for (const name of Object.keys(objectify(config.rules))) {
+        combinedConfig.rules[name] = config.rules[name];
     }
+
+    return combinedConfig;
```

Comment:
Sorry. take the `_.defaults` idea back. Just realize I fall into the generalization trap. (slapping my hand).

Each property: `rules`, `rulesDirectory` and `extend` should be handled differently. Ordering in `rulesDirectory` doesn't matter, but it does in `extend`.